curvefs/src/client/common/config.cpp

```diff
@@ -243,6 +244,53 @@ void InitKVClientManagerOpt(Configuration *conf,
                               &config->getThreadPooln);
 }
 
+void InitFileSystemOption(Configuration* c, FileSystemOption* option) {
+    c->GetValueFatalIfFail("fs.cto", &option->cto);
+    c->GetValueFatalIfFail("fs.cto", &FLAGS_enableCto);
+    c->GetValueFatalIfFail("fs.disableXattr", &option->disableXattr);
+    c->GetValueFatalIfFail("fs.maxNameLength", &option->maxNameLength);
+    c->GetValueFatalIfFail("fs.accessLogging", &FLAGS_access_logging);
+    {  // kernel cache option
+        auto o = &option->kernelCacheOption;
+        c->GetValueFatalIfFail("fs.kernelCache.attrTimeout",
+                               &o->attrTimeout);
+        c->GetValueFatalIfFail("fs.kernelCache.dirAttrTimeout",
+                               &o->dirAttrTimeout);
+        c->GetValueFatalIfFail("fs.kernelCache.entryTimeout",
+                               &o->entryTimeout);
+        c->GetValueFatalIfFail("fs.kernelCache.dirEntryTimeout",
```

Comment:
> if fs.kernelCache.entryTimeout=true, fs.kernelCache.dirEntryTimeout=false, dir entry will not cache, but file entry will be cached?

yep. 

> if fs.kernelCache.entryTimeout=false, fs.kernelCache.dirEntryTimeout=false, only dir entry cached?

yep.

Reply:
> what's the diffrence between fs.kernelCache.dirEntryTimeout and fs.kernelCache.entryTimeout?

This designed for assgin different timeout for different types of entris. If the entry is file, we will reply to kernel with timeout ` fs.kernelCache.entryTimeout`, otherwise reply with `fs.kernelCache.dirEntryTimeout`.

```
void FileSystem::SetEntryTimeout(EntryOut* entryOut) {
    auto option = option_.kernelCacheOption;
    if (IsDir(entryOut->attr)) {
        entryOut->entryTimeout = option.dirEntryTimeout;
        entryOut->attrTimeout = option.dirAttrTimeout;
    } else {
        entryOut->entryTimeout = option.entryTimeout;
        entryOut->attrTimeout = option.attrTimeout;
    }
}
```